### PR TITLE
Telebaton fix, refactor and documentation 

### PR DESCRIPTION
**Pull request by dearmochi**
## Context

**Baton here means the telescopic kind, not stun baton**

Batons when used for a non-lethal attack (knockdown) do not trigger an attack animation on the user like other melee weapons do. 

The code is also of poor quality.

## What Does This PR Do

- **Knocking down a mob using a baton plays the attack animation on the user**
- Refactors, cleans and [documents](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md) baton code
- Moves the sole ban hammer proc to the same file as the class definition (`code\game\objects\items\weapons\weaponry.dm`)
- Renames `code\game\objects\items\weapons\swords_axes_etc.dm` to `code\game\objects\items\weapons\batons.dm`.

## Why It's Good For The Game

Better code quality and visual feedback on attack. 

## Changelog
:cl:
add: Add attack animation when knocking down with telescopic baton and similar
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/batons.dm [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		else
+			user.take_organ_damage(force * 2)
+		return
+	if(isrobot(target))
+		return ..()
+	if(!isliving(target))
+		return
+
+	if(user.a_intent == INTENT_HARM) // Lethal attack
+		if(!..() || !isrobot(target))
```
Comment (farie82) [inline code/game/objects/items/weapons/batons.dm:47]: isrobot can never be true here.
This is best to be rewritten as `return ..()`
How it's written currently is a bit odd

File: code/game/objects/items/weapons/batons.dm [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+/obj/item/melee/classic_baton/attack(mob/target, mob/living/user)
+	if(!on)
+		return ..()
+
+	add_fingerprint(user)
+	if((CLUMSY in user.mutations) && prob(50))
+		user.visible_message("<span class='danger'>[user] accidentally clubs [user.p_them()]self with [src]!</span>", \
+							"<span class='userdanger'>You accidentally club yourself with [src]!</span>")
+		user.Weaken(force * 3)
```
Comment (farie82) [inline code/game/objects/items/weapons/batons.dm:34]: Holy balls that weaken here

File: code/game/objects/items/weapons/batons.dm [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	item_state = "classic_baton"
+	slot_flags = SLOT_BELT
+	force = 12 //9 hit crit
+	w_class = WEIGHT_CLASS_NORMAL
+	/// Whether the baton is on cooldown
+	var/on_cooldown = FALSE
+	/// Whether the baton is toggled on (to allow attacking)
+	var/on = TRUE
+
+/obj/item/melee/classic_baton/attack(mob/target, mob/living/user)
```
Comment (farie82) [inline code/game/objects/items/weapons/batons.dm:26]: The target should always be a living here

File: code/game/objects/items/weapons/batons.dm [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		user.Weaken(force * 3)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.apply_damage(force * 2, BRUTE, "head")
+		else
+			user.take_organ_damage(force * 2)
+		return
+	if(isrobot(target))
+		return ..()
+	if(!isliving(target))
```
Comment (farie82) [inline code/game/objects/items/weapons/batons.dm:43]: Can never happen

Comment (dearmochi) [reply code/game/objects/items/weapons/batons.dm]: All donezo

**Pull request merged**